torch/_inductor/codecache.py

```diff
@@ -454,8 +458,12 @@ def write_atomic(
         path.parent.mkdir(parents=True, exist_ok=True)
     tmp_path = path.parent / f".{os.getpid()}.{threading.get_ident()}.tmp"
     write_mode = "w" if isinstance(content, str) else "wb"
-    with tmp_path.open(write_mode) as f:
-        f.write(content)
+    if encode_utf_8:
+        with tmp_path.open(write_mode, encoding="utf-8") as f:
```

Comment:
Cleaner to use a single context but move the condition to `encoding`.

Reply:
@desertfire Good suggestion, done.